jca/WFLY-13715_Certify_Oracle_19c_RAC_Cluster_Support_with_Oracle_UCP.adoc

```diff
@@ -0,0 +1,51 @@
+= WFLY-13715 Certify Oracle 19c RAC Cluster Support with Oracle UCP
+:author:            Tomasz Adamski
+:email:             tadamski@redhat.com
+:toc:               left
+:icons:             font
+:keywords:          thread pool,Oracle UCP
+:idprefix:
+:idseparator:       -
+:issue-base-url:    https://issues.redhat.com/browse/
+
+== Overview
+
+One of our customer's application relies on Oracle RAC, which further relies on Oracle's Universal Connection Pool (UCP). UCP manages it's own connection pool and provides it using PoolDataSource. Usage of UCP allows for a number of advantages (fast connection failover, run time connection loadbalancing, connection affinity) which are required in customer's use case. 
+On the other hand, Iron Jacamar manages it's own connection pool and was not designed for integration with DataSources with their own pool. Furthermore, there are no plans of changing IronJacamar architecture for such use case. As a result, the goal of this RFE is to provide workaround that would allow the customer to take advantage of most of UCP features without changing IronJacamar architecture.
+The workaround that allows to achieve this is a notification send to Oracle's specific connection implementation when it is started and ended.
+
+== Issue Metadata
+
+=== Issue:
+
+* {issue-base-url}/WFLY-13715[WFLY-13715]
+
+=== Related Issues:
+
+* {issue-base-url}/EAP7-168[EAP7-168]
+
+=== Dev Contacts:
+
+* mailto:{email}[{author}]
+
+=== QE Contacts:
+
+*
+
+=== Affected Projects or Components:
+
+* IronJacamar, WildFly
+
+== Requirements
+
+* customers should be able to take advantage of Oracle UCP features
+* there would be no changes to IronJacamar architecture
+
+== Proposed solution
+* notify OracleConnection about start and end of the request
+* a system property (notify.oracle.connecton) should be set that will switch the notification on
+* if notification is turned on, in org.jboss.jca.core.connectionmanager.pool.mcp.SemaphoreArrayListManagedConnectionPool invoke oracle.jdbc.OracleConnection#beginRequest,oracle.jdbc.OracleConnection#endRequest when connection is started and ended
+
```

Comment:
We should not rely on Oracle specific implementation. We can use the equivalent method of standard java.sqlconnection for Java9+ and require customers to use Java9+
Moreover, considering we are using a standard method, we can even avoid to have system property and just check through reflection the presence of java.sql.Connection#beginRequest and java.sql.Connection#endRequest methods and invoke them IIF they are present (aka we are using JDBC API of Java9+)

Reply:
fixed